### DISCUSSION TITLE
fail to start teleport : Failed to resolve tunnel address

### DISCUSSION THREAD

**Discussion by DyO63** [gravitational/teleport#12204] [Q&A]
Hi,

i just set up a fresh goteleport server on debian 11 using the link 
https://goteleport.com/docs/installation/

As it is, at the moment for POC, i created a selfsigned certificate
All seems alright

To test it, i set up a debian 11 client using the link 
https://goteleport.com/docs/server-access/getting-started/

After succeed generating token, i tried the command
`teleport start \
   --roles=node \
   --token=/xxx/token.file \
   --auth-server=myserver:443`

But i always have errors : 

`
INFO [PROC:1]    Connecting to the cluster myserver with TLS client certificate. service/connect.go:164
ERRO [PROC:1]    Failed to resolve tunnel address Get "https://myserver:443/webapi/find": x509: certificate is not valid for any names, but wanted to match myserver reversetunnel/transport.go:90
`
When i launch a 
"tctl status" on the server, i just have a selfsigned certificat warning"

and then : 
`
Cluster  myserver
Version  9.1.1
Host CA  never updated
User CA  never updated
Jwt CA   never updated
CA pin   shaxxxxxxxxx
`

Could it be (the certificate)  the root of my problem ?

Thanks

D.

**Comment by webvictim** [comment]
How did you create your self-signed certiifcate? It looks like it doesn't have any SANs (subject alternative names) set, which will cause an issue.

If you leave `https_keypairs` blank in the Teleport config, Teleport will generate an appropriate self-signed certificate that you can use.

**Comment by DyO63** [comment]
Hi,

thanks for your reply.

I tried to delete the 2 lines under https_keypairs.
Restart teleport service.
Things are little bit differents:  
`
INFO Joining the cluster with a secure token. service/connect.go:395
[PROC:1]    ERRO Node failed to establish connection to cluster: Post "https://myserver:443/v1/webapi/host/credentials": x509: certificate signed by unknown authority, invalid character '<' looking for beginning of value. service/connect.go:86
` 

What did i miss ?

I don't know where there could be a missing character.

Thanks

**Reply by webvictim** [reply]
If you're using a self-signed certificate you'll need to use `teleport start --insecure` when connecting your agents.

**Comment by DyO63** [comment]
Thanks again, but doesn't work

Always have the same message  " invalid character '<' "

i check my teleport.yaml et don't see any invalid character

Could it be an other conf file to check ?

**Comment by webvictim** [comment]
What do you see when you run `curl -v https://myserver:443/webapi/ping`?

**Comment by DyO63** [comment]
`curl: (60) SSL certificate problem: self signed certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.`

damned...
That's the problem.

I made a csr to generate a valid certificate

wait and see

Thanks a lot

**Comment by DyO63** [comment]
That works !

I conigured my officiel certificate and it works.

Thanks à lot
